### DISCUSSION TITLE
Calculating a CAC estimate

### DISCUSSION THREAD

**Discussion by braggbear** [sibylhe/mmm_stan#9] [Q&A]
The blog post discusses how to calculate ROAS and mROAS.  But what if you want to calculate CAC (cost per acquisition) and mCAC (marginal cost per acquisition)?  Thanks.

**Comment by sibylhe** [comment]
I think MMM cannot address this... CAC needs individual-level tracking, depends on if you can track the number of new customers. If you can give an example of your problem and data, I might have a more specific answer.

**Reply by braggbear** [reply]
Thank you for your response. CAC is similar to ROAS but it's sales instead of revenue. 

In a simple MMM with no adstock or diminishing returns (ie: sales ~ B0 + B1*media_spend), CAC is just the inverse of B1. In other words, if B1 is how sales go up when you spend one more dollar, then the inverse of that is how much you have to spend for one sale (which is CAC or CPA). 

But i'm not sure how to calculate the CAC for an MMM model like the one we are working with that includes adstock and hill parameters.  I don't think we can simply take the inverse of the media betas.

**Comment by sibylhe** [comment]
Forget about my first reply, I thought CAC is the average cost for acquiring a new user.

The target variable "sales", is it the number of orders?
CAC = media spend / media contribution (how many orders this channel bring)

If the above understanding is correct, CAC is the inverse of ROAS, since ROAS = media contribution / media spend. mCAC is the inverse of mROAS.
So quick solution is to take the inverse of ROAS, just one more step.

If you're interested in "how":
This repo is about multiplicative MMM: sales = B0 * media_impression^B1
To calculate CAC, you need two components (same as ROAS):
1) media contribution (no matter by revenue or by orders, goes with your target variable): see section 2.2 [Decompose sales to media channels' contribution](https://github.com/sibylhe/mmm_stan#decompose-sales-to-media-channels-contribution), implementation is the [mmm_decompose_contrib](https://github.com/sibylhe/mmm_stan/blob/main/mmm_stan.py#L483) function  
2) media spend: adstocked and hill-transformed, [line 797-802](https://github.com/sibylhe/mmm_stan/blob/main/mmm_stan.py#L797). 

Finally, you may run the roas functions ([line 805-829](https://github.com/sibylhe/mmm_stan/blob/main/mmm_stan.py#L805)), which takes the above 2 components, and take the inverse of the result. Or you may modify the roas functions to calculate cac.

**Reply by braggbear** [reply]
Thank you. This is helpful!

A couple questions on the intuition of the calculation: 

(1) What do x1 and x2 represent?  For example, when X1=100, what is the 100?  

(2) If I want to calculate ROAS for channel 1, would I just take the 283.23 (the media contribution) and divide it by the average spend for that channel?  Or do I divide it by some transformation of the spend?

I know you provided the python code but trying to understand the intuition since my code is in R.  =)

Thanks for your input!

**Reply by sibylhe** [reply]
(1) What do x1 and x2 represent? For example, when X1=100, what is the 100?
They are media variables - channel impression (or spend) of a week.  x1=100 -> this week TV impression is 100

(2) 283.23 is a channel's media contribution of one week. 
Spend needs to be adstocked and hill-transformed, i.e., transformed_spend = hill_transform(adstock(raw_spend))

Let's say channel 1's 4 weeks' raw spend: [100, 150, 100, 200]
transformed_spend: [70, 170, 120, 180]
media_contribution: [283, 650, 420, 670]
ROAS of the 1st week = 283/70
ROAS of the 4 weeks = sum([283, 650, 420, 670])/sum([70, 170, 120, 180])
